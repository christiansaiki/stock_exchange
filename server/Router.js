import { RequestHandler } from '../controllers';

/**
 * Router - this class a method that routes all the requests
 * it relies on express.Router() in order to create API versioning (like v1, v2 and so on)
 *
 * @export
 * @class Router
 */
export default class Router {
	constructor(fastify, mongo) {
		this.fastify = fastify;
		this.mongo = mongo;
	}

	init() {
		const { fastify, mongo } = this;
		const handler = new RequestHandler(fastify, mongo);

		fastify.get('/exchange',  handler.get);
	}
}
