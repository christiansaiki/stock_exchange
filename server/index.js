const fastify = require('fastify')();
import Router from './Router';
import { MongoDB } from '../db';

// Environment Variables
const env = process.env.NODE_ENV || 'local';
const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/stock_exchange';

// Init Mongo
const mongo = new MongoDB(env, mongoUrl);
mongo.init();

// Init Router
const router = new Router(fastify, mongo);
router.init();

// handle rejections
process.on('unhandledRejection', (reason, p) => {
	console.log(reason);
	console.log(p);
});

export default {
	mongo,
	app: fastify
};