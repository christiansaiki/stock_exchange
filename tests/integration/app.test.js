import server from '../../server';
const populate = require('../../db/populate');

describe('APP Testing', () => {
	beforeAll(() => {
		return populate(server.mongo);
	});

	afterAll((done) => {
		populate(server.mongo, true);
		server.app.close(done);
	});

	test('GET `/` route', done => {
		server.app.inject({
			method: 'GET',
			url: '/'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - return C1 - US, Automobile', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=US&Category=Automobile&BaseBid=10'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - return C2 - US, IT', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=US&Category=IT&BaseBid=30'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - return C3 - RU, IT', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=RU&Category=IT&BaseBid=10'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - return C2 - IN, Finance', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=IN&Category=Finance&BaseBid=30'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - No targeting - BR, Finance', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=BR&Category=Finance&BaseBid=30'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - No Budget - US, Finance', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=US&Category=Finance&BaseBid=210'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});

	test('GET `/exchange` route - No Base Bid - US, Finance', done => {
		server.app.inject({
			method: 'GET',
			url: '/exchange?countrycode=US&Category=Finance&BaseBid=9'
		}, (err, response) => {
			expect(err).toMatchSnapshot();
			expect(response.statusCode).toMatchSnapshot();
			expect(response.payload).toMatchSnapshot();
			done();
		});
	});
});
