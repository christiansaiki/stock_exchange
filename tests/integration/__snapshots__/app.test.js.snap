// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP Testing GET \`/\` route 1`] = `null`;

exports[`APP Testing GET \`/\` route 2`] = `404`;

exports[`APP Testing GET \`/exchange\` route - No Base Bid - US, Finance 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - No Base Bid - US, Finance 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - No Base Bid - US, Finance 3`] = `"No Companies Passed from BaseBid check"`;

exports[`APP Testing GET \`/exchange\` route - No Budget - US, Finance 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - No Budget - US, Finance 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - No Budget - US, Finance 3`] = `"No Companies Passed from Budget"`;

exports[`APP Testing GET \`/exchange\` route - No targeting - BR, Finance 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - No targeting - BR, Finance 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - No targeting - BR, Finance 3`] = `"No Companies Passed from Targeting"`;

exports[`APP Testing GET \`/exchange\` route - return C1 - US, Automobile 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - return C1 - US, Automobile 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - return C1 - US, Automobile 3`] = `"C1"`;

exports[`APP Testing GET \`/exchange\` route - return C2 - IN, Finance 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - return C2 - IN, Finance 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - return C2 - IN, Finance 3`] = `"C2"`;

exports[`APP Testing GET \`/exchange\` route - return C2 - US, IT 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - return C2 - US, IT 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - return C2 - US, IT 3`] = `"C2"`;

exports[`APP Testing GET \`/exchange\` route - return C3 - RU, IT 1`] = `null`;

exports[`APP Testing GET \`/exchange\` route - return C3 - RU, IT 2`] = `200`;

exports[`APP Testing GET \`/exchange\` route - return C3 - RU, IT 3`] = `"C3"`;
